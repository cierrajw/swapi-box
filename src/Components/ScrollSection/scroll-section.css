body{
  background-color: #7789a5;
}

.component-container{
  display: grid;
  grid-template-columns: 1.5fr 4fr;
  height: 100vh;
}

.right-section{
  display: grid;
  grid-template-rows: .5fr .7fr 5fr;
}

.scroll-section{
  background-color: blue;
  grid-column: 1;
  grid-row: 1/3;
  height: 100vh;
  color: gold;
}

.scroll-text{
    color: gold;
    overflow-y: visible;
    font-size:30px;
    text-align:center;
    font-family:sans-serif;
    font-size: 2rem;
    animation: autoscroll 1000s linear;
}

/* @keyframes autoscroll{
    to{
        margin-top:-50000px;
    }
} */
