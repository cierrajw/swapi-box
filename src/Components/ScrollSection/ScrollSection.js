import React, { Component } from 'react';
import { shallow, mount } from 'enzyme';
import './scroll-section.css'

class ScrollSection extends Component{
  constructor(){
    super();

    this.state={
      filmText: ''
    }
  }

  componentDidMount = () => {
    this.displayFilmText()
    console.log(this.state.filmText)
  }

  displayFilmText = () => {
    try {
      fetch('https://swapi.co/api/films/')
        .then(response => response.json())
        .then(starWarsData => this.getFilm(starWarsData.results))
    } catch(error) {
      console.log(error)
    }
  }

  getFilm = (films) => {
    const filmScrolls = films.map((film, index) => {
      return film.opening_crawl
    })
    const randomScroll = filmScrolls[Math.floor(Math.random() * filmScrolls.length + 1)]
    this.setState({filmText: randomScroll})
  }

  render(){
    return(
      <div className="swapi-text-div">
        <h1 className="swapi-title">Swapi Boy</h1>
        <section className="filmtext-container">
          <div className="film-text">{this.state.filmText}</div>
        </section>
      </div>
    )
  }
}

export default ScrollSection;
