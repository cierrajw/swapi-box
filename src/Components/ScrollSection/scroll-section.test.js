import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import ScrollSection from './ScrollSection.js';

describe('Scroll Section', () => {

  it('should match the snapshot', () => {

    const wrapper = shallow(<ScrollSection />);

    expect(wrapper).toMatchSnapshot();

  });

});
