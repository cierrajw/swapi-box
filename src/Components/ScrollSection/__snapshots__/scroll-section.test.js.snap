// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll Section should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollSection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="swapi-title"
        >
          Swapi Box
        </h1>,
        <section
          className="filmtext-container"
        >
          <div
            className="film-text"
          >
            
          </div>
        </section>,
      ],
      "className": "swapi-text-div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Swapi Box",
          "className": "swapi-title",
        },
        "ref": null,
        "rendered": "Swapi Box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="film-text"
          >
            
          </div>,
          "className": "filmtext-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "film-text",
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="swapi-title"
          >
            Swapi Box
          </h1>,
          <section
            className="filmtext-container"
          >
            <div
              className="film-text"
            >
              
            </div>
          </section>,
        ],
        "className": "swapi-text-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swapi Box",
            "className": "swapi-title",
          },
          "ref": null,
          "rendered": "Swapi Box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="film-text"
            >
              
            </div>,
            "className": "filmtext-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "film-text",
            },
            "ref": null,
            "rendered": "",
            "type": "div",
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
